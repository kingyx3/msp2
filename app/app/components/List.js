import React from "react";
import { View, TouchableHighlight, StyleSheet, } from "react-native";
import Swipeable from "react-native-gesture-handler/Swipeable";
import { Badge } from 'react-native-elements';
import {
  AntDesign,
  EvilIcons,
  MaterialIcons,
  MaterialCommunityIcons,
} from "@expo/vector-icons";
import styled from "styled-components/native";
import * as Typography from "../config/Typography";
import colors from "../config/colors";
import ImgSliderItems from "./ImgSliderItems";

export const Default = ({ containedicon, icon, secondary, meta, title, onPress, ...otherProps }) => {
  return (
    <TouchableHighlight {...otherProps} underlayColor={colors.lightgray} onPress={onPress}>
      <Container >
        {containedicon && (
          <IconContainer >
            <MaterialIcons name={containedicon} size={18} />
          </IconContainer>
        )}
        <TextContainer>
          <Typography.Sub1 color={colors.black}>{title}</Typography.Sub1>
          {meta && <Typography.SP color={colors.black}>{meta}</Typography.SP>}
          {secondary && <Typography.P>{secondary}</Typography.P>}
        </TextContainer>
        {icon && (
          <MaterialCommunityIcons
            name={icon}
            size={24}
            style={{ color: colors.gray }}
          />
        )}
      </Container>
    </TouchableHighlight>
  );
};

export const UserList = ({
  image,
  imagesmall,
  imagexsmall,
  meta,
  onPress,
  secondary,
  title,
  RightActions,
  unreadCount,
  ...otherProps
}) => {
  return (
    <Swipeable renderRightActions={RightActions}>
      <TouchableHighlight {...otherProps} underlayColor={colors.faintgray} onPress={onPress}>
        <View>
          <Container>
            {image ? (
              <ImgSliderItems
                source={image}
                customStyle={{
                  width: 70,
                  height: 70,
                  borderRadius: 35,
                  marginRight: 15,
                }}
                placeholder={require('../assets/default_user.png')}
              />
              // <Avatar source={{ uri: image }} />
            ) : imagesmall ? (
              <AvatarSmall source={{ uri: imagesmall }} />
            ) : imagesmall === null ? (
              <AvatarSmallEmpty>
                <AntDesign name="user" size={26} color="white" />
              </AvatarSmallEmpty>
            ) : imagexsmall ? (
              <AvatarXsmall source={{ uri: imagexsmall }} />
            ) : imagexsmall === null ? (
              <AvatarXsmallEmpty>
                <AntDesign name="user" size={26} color="white" />
              </AvatarXsmallEmpty>
            ) : (
              <AvatarEmpty>
                <AntDesign name="user" size={26} color="white" />
              </AvatarEmpty>
            )}
            <TextContainer>
              <FirstLine>
                <Typography.Sub1 color={colors.black}>{title}</Typography.Sub1>
                <Typography.Cap color={colors.black}>{meta}</Typography.Cap>
              </FirstLine>
              <FirstLine>
                {secondary && (
                  <Typography.P numberOfLines={1} color={colors.gray}>
                    {secondary}
                  </Typography.P>
                )}
                {unreadCount > 0 &&
                  <Badge
                    value={unreadCount}
                    status="primary"
                    size="medium"
                    containerStyle={{ paddingRight: 10 }}
                  />
                }
              </FirstLine>
            </TextContainer>
          </Container>
        </View>
      </TouchableHighlight>
    </Swipeable>
  );
};

export const LogsList = ({
  image,
  imagesmall,
  imagexsmall,
  meta,
  onPress,
  secondary,
  title,
  RightActions,
  unreadCount,
  defaultImageUrl,
  created,
  message,
  logType,
  amount,
  positive,
  testID
}) => {
  return (
    <TouchableHighlight underlayColor={colors.faintgray} onPress={onPress}>
      <View style={styles.item2}>
        <View style={styles.header}>
          <Typography.H4 testID={testID} bold>{title}</Typography.H4>
          <Typography.Cap>{created}</Typography.Cap>
        </View>
        <View style={styles.details}>
          <Typography.P>{message}</Typography.P>
          <Typography.P color={positive ? 'green' : 'red'} bold>{amount}</Typography.P>
        </View>
        <Typography.P numberOfLines={1} color={colors.gray}>{secondary}</Typography.P>
      </View>
    </TouchableHighlight>
  );
};

const Container = styled.View`
  background-color: white;
  flex-direction: row;
  align-items: center;
  padding: 20px 0;
`;

const TextContainer = styled.View`
  flex: 1;
  flex-direction: column;
`;

const IconContainer = styled.View`
  justify-content: center;
  align-items: center;
  width: 35px;
  height: 35px;
  background-color: ${colors.faintgray};
  border-radius: 8px;
  border: 1px solid ${colors.lightgray};
  margin-right: 15px;
`;

const Avatar = styled.Image`
  width: 70px;
  height: 70px;
  border-radius: 35px;
  margin-right: 15px;
`;

const AvatarEmpty = styled.View`
  justify-content: center;
  align-items: center;
  width: 70px;
  height: 70px;
  border-radius: 35px;
  margin-right: 15px;
  background-color: ${colors.lightgray};
`;

const AvatarSmall = styled.Image`
  width: 46px;
  height: 46px;
  border-radius: 23px;
  margin-right: 12px;
  background-color: ${colors.lightgray};
`;

const AvatarSmallEmpty = styled.View`
  display: flex;
  justify-content: center;
  align-items: center;
  width: 46px;
  height: 46px;
  border-radius: 23px;
  margin-right: 12px;
  background-color: ${colors.lightgray};
`;

const AvatarXsmall = styled.Image`
  width: 34px;
  height: 34px;
  border-radius: 17px;
  margin-right: 10px;
  background-color: ${colors.lightgray};
`;

const AvatarXsmallEmpty = styled.View`
  display: flex;
  justify-content: center;
  align-items: center;
  width: 34px;
  height: 34px;
  border-radius: 17px;
  margin-right: 10px;
  background-color: ${colors.lightgray};
`;

const FirstLine = styled.View`
  flex-direction: row;
  justify-content: space-between;
`;

const styles = StyleSheet.create({
  container: {
    flex: 1,
    marginTop: 20,
  },
  item: {
    backgroundColor: '#f9c2ff',
    padding: 20,
    marginVertical: 8,
    marginHorizontal: 16,
  },
  bookingId: {
    fontSize: 16,
    fontWeight: 'bold',
  },
  bookingDate: {
    fontSize: 14,
  },
  hostEarnings: {
    fontSize: 14,
    fontStyle: 'italic',
  },



  header: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
    marginBottom: 4
  },
  item2: {
    backgroundColor: '#fff3ff',
    paddingVertical: 8,
    paddingHorizontal: 16,
    marginVertical: 4,
    borderRadius: 8,
  },

  details: {
    flexDirection: 'row',
    alignItems: 'center',
    marginBottom: 4,
    justifyContent: 'space-between'
  },
});