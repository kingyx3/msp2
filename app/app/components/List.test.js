import React from 'react';
import { render } from '@testing-library/react-native';
import { Default, UserList, LogsList } from './List';
import renderer from 'react-test-renderer';

jest.mock('./ImgSliderItems', () => 'ImgSliderItems');

describe('Default Component', () => {
  it('renders with title and secondary text', () => {
    const { getByText } = render(
      <Default title="Test Title" secondary="Test Secondary" />
    );
    expect(getByText('Test Title')).toBeTruthy();
    expect(getByText('Test Secondary')).toBeTruthy();
  });

  test('renders icon correctly - Default', () => {
    const tree = renderer.create(
      <Default title="Test Title" icon="camera" />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  // Additional tests for other prop combinations...
});

describe('UserList Component', () => {
  it('renders correctly with image and text props', () => {
    const { getByText } = render(
      <UserList
        title="User Title"
        secondary="User Secondary"
        image="http://example.com/image.jpg"
      />
    );

    expect(getByText('User Title')).toBeTruthy();
    expect(getByText('User Secondary')).toBeTruthy();
    // Additional checks for image rendering...
  });

  test('renders icon correctly - UserList', () => {
    const tree = renderer.create(
      <UserList
        title="User Title"
        secondary="User Secondary"
        image="http://example.com/image.jpg"
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
  // Additional tests...
});

describe('LogsList Component', () => {
  it('renders correctly with log data', () => {
    const { getByText } = render(
      <LogsList
        title="Log Title"
        message="Log Message"
        secondary="Log Secondary"
        amount="$100"
      />
    );

    expect(getByText('Log Title')).toBeTruthy();
    expect(getByText('Log Message')).toBeTruthy();
    expect(getByText('Log Secondary')).toBeTruthy();
    expect(getByText('$100')).toBeTruthy();
  });

  test('renders icon correctly - LogsList', () => {
    const tree = renderer.create(
      <LogsList
        title="Log Title"
        message="Log Message"
        secondary="Log Secondary"
        amount="$100"
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
  // Additional tests...
});
