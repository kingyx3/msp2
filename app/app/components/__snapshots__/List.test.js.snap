// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Default Component renders icon correctly - Default 1`] = `
<View
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "white",
        "flexDirection": "row",
        "paddingBottom": 20,
        "paddingLeft": 0,
        "paddingRight": 0,
        "paddingTop": 20,
      }
    }
  >
    <View
      style={
        {
          "flexBasis": 0,
          "flexDirection": "column",
          "flexGrow": 1,
          "flexShrink": 1,
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
          }
        }
      >
        <View
          style={{}}
        >
          <Text
            style={
              {
                "fontFamily": "Avenir",
              }
            }
          >
            <Text>
              <Text
                style={
                  [
                    {
                      "fontSize": 7.2,
                    },
                    {
                      "color": "#333333",
                    },
                  ]
                }
              >
                Test Title
              </Text>
            </Text>
          </Text>
        </View>
      </View>
    </View>
    <Text />
  </View>
</View>
`;

exports[`LogsList Component renders icon correctly - LogsList 1`] = `
<View
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
>
  <View
    style={
      {
        "backgroundColor": "#fff3ff",
        "borderRadius": 8,
        "marginVertical": 4,
        "paddingHorizontal": 16,
        "paddingVertical": 8,
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginBottom": 4,
        }
      }
    >
      <View
        style={{}}
      >
        <Text
          style={
            {
              "fontFamily": "Avenir",
            }
          }
        >
          <Text
            style={
              {
                "fontWeight": "bold",
              }
            }
          >
            <Text
              style={
                [
                  {
                    "fontSize": 8.1,
                  },
                  {
                    "color": undefined,
                  },
                ]
              }
            >
              Log Title
            </Text>
          </Text>
        </Text>
      </View>
      <View
        style={{}}
      >
        <Text
          style={
            {
              "fontFamily": "Avenir",
            }
          }
        >
          <Text
            style={
              [
                {
                  "fontSize": 5.4,
                },
                {
                  "color": undefined,
                },
              ]
            }
          />
        </Text>
      </View>
    </View>
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginBottom": 4,
        }
      }
    >
      <View
        style={{}}
      >
        <Text
          style={
            {
              "fontFamily": "Avenir",
            }
          }
        >
          <Text>
            <Text
              style={
                [
                  {
                    "fontSize": 6.3,
                    "lineHeight": 22,
                  },
                  {
                    "color": undefined,
                  },
                ]
              }
            >
              Log Message
            </Text>
          </Text>
        </Text>
      </View>
      <View
        style={{}}
      >
        <Text
          style={
            {
              "fontFamily": "Avenir",
            }
          }
        >
          <Text
            style={
              {
                "fontWeight": "bold",
              }
            }
          >
            <Text
              style={
                [
                  {
                    "fontSize": 6.3,
                    "lineHeight": 22,
                  },
                  {
                    "color": "red",
                  },
                ]
              }
            >
              $100
            </Text>
          </Text>
        </Text>
      </View>
    </View>
    <View
      style={{}}
    >
      <Text
        numberOfLines={1}
        style={
          {
            "fontFamily": "Avenir",
          }
        }
      >
        <Text>
          <Text
            style={
              [
                {
                  "fontSize": 6.3,
                  "lineHeight": 22,
                },
                {
                  "color": "#707070",
                },
              ]
            }
          >
            Log Secondary
          </Text>
        </Text>
      </Text>
    </View>
  </View>
</View>
`;

exports[`UserList Component renders icon correctly - UserList 1`] = `
<View
  collapsable={false}
  handlerTag={3}
  handlerType="PanGestureHandler"
  onGestureHandlerEvent={[Function]}
  onGestureHandlerStateChange={[Function]}
  onLayout={[Function]}
  style={
    {
      "overflow": "hidden",
    }
  }
>
  <View
    collapsable={false}
    handlerTag={4}
    handlerType="TapGestureHandler"
    onGestureHandlerEvent={[Function]}
    onGestureHandlerStateChange={[Function]}
    pointerEvents="auto"
    style={
      {
        "transform": [
          {
            "translateX": -0,
          },
        ],
      }
    }
  >
    <View
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
    >
      <View>
        <View
          style={
            {
              "alignItems": "center",
              "backgroundColor": "white",
              "flexDirection": "row",
              "paddingBottom": 20,
              "paddingLeft": 0,
              "paddingRight": 0,
              "paddingTop": 20,
            }
          }
        >
          <ImgSliderItems
            customStyle={
              {
                "borderRadius": 35,
                "height": 70,
                "marginRight": 15,
                "width": 70,
              }
            }
            placeholder={1}
            source="http://example.com/image.jpg"
          />
          <View
            style={
              {
                "flexBasis": 0,
                "flexDirection": "column",
                "flexGrow": 1,
                "flexShrink": 1,
              }
            }
          >
            <View
              style={
                {
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                }
              }
            >
              <View
                style={{}}
              >
                <Text
                  style={
                    {
                      "fontFamily": "Avenir",
                    }
                  }
                >
                  <Text>
                    <Text
                      style={
                        [
                          {
                            "fontSize": 7.2,
                          },
                          {
                            "color": "#333333",
                          },
                        ]
                      }
                    >
                      User Title
                    </Text>
                  </Text>
                </Text>
              </View>
              <View
                style={{}}
              >
                <Text
                  style={
                    {
                      "fontFamily": "Avenir",
                    }
                  }
                >
                  <Text
                    style={
                      [
                        {
                          "fontSize": 5.4,
                        },
                        {
                          "color": "#333333",
                        },
                      ]
                    }
                  />
                </Text>
              </View>
            </View>
            <View
              style={
                {
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                }
              }
            >
              <View
                style={{}}
              >
                <Text
                  numberOfLines={1}
                  style={
                    {
                      "fontFamily": "Avenir",
                    }
                  }
                >
                  <Text>
                    <Text
                      style={
                        [
                          {
                            "fontSize": 6.3,
                            "lineHeight": 22,
                          },
                          {
                            "color": "#707070",
                          },
                        ]
                      }
                    >
                      User Secondary
                    </Text>
                  </Text>
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
